src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1369,12 +1369,42 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
                         }
                         else
                         {
+                            SourceLocalSymbol sourceLocal;
+                            type = null;
+
+                            if (node.SyntaxTree == localSymbolLocation.SourceTree && (object)(sourceLocal = localSymbol as SourceLocalSymbol) != null &&
+                                sourceLocal.IdentifierToken.Parent?.Kind() == SyntaxKind.Argument)
+                            {
+                                var argument = (ArgumentSyntax)sourceLocal.IdentifierToken.Parent;
+
+                                if (argument.Identifier == sourceLocal.IdentifierToken &&
+                                    argument.Type.IsVar)
+                                {
+                                    // We are referring to an out variable which might need type inference.
+                                    // If it is in fact needs inference, it is illegal to reference it in the same argument list. 
+                                    // Technically, we could support some restricted scenarios (when it is used as another argument in the same list), 
```

Comment:
Please be specific and point to the relevant places in this PR. Thanks.

---
In reply to: [64249559](https://github.com/dotnet/roslyn/pull/11493#discussion_r64249559) [](ancestors = 64249559,64161065)